### PR TITLE
[Core] Adding analytic direction to AssignVectorByDirectionToConditionProcess

### PR DESCRIPTION
**Pull request by loumalouomega**
This adds the option to add an analytic direction to the conditions and nodes

Makes the code more pythonic and cleans up the test

### CODE REVIEW

**Comment by philbucher** [comment]
could you please briefly explain what you are doing here?
I don't get it :)

**Comment by loumalouomega** [comment]
> could you please briefly explain what you are doing here?
> I don't get it :)

Additionally to allowing to consider an analytic function for the modulus, this allows to consider an analytic direction

**Comment by loumalouomega** [comment]
@philbucher are you happy with the explanation or do you need more details?. Basically you can define a function for the direction (but it is assumed to be unidirectional). This string is appended to the function of the modulus

File: kratos/python_scripts/assign_vector_by_direction_to_condition_process.py [hunk 1]
```diff
@@ -71,45 +93,60 @@ def __init__(self, Model, settings ):
...
+            # Normalize direction
+            if all_numeric:
+                direction_norm = math.sqrt(direction_norm)
+                if direction_norm < 1.0e-6:
+                    raise Exception("Direction norm is close to 0 in AssignVectorByDirectionToConditionProcess.")
 
+                for i in range(0,3):
+                    unit_direction[i] = unit_direction[i]/direction_norm
+            else:
+                KratosMultiphysics.Logger.PrintWarning("AssignVectorByDirectionToConditionProcess", "You are defining a direction as string. Unitary definition will be assumed")
```
Comment (philbucher) [inline kratos/python_scripts/assign_vector_by_direction_to_condition_process.py]: why the warning?

**Review by philbucher — COMMENTED** [review]
the `assign_scalar_variable_process` and the `assign_scalar_variable_to_conditions_process` can handle the string-defined direction?

With all the cosmetic changes I can barely follow/understand the changes. Can you please make a PR with only the cosmetic changes? This is an important process, we have to be careful
In general, you can request my review with cosmetic-only PRs, I would approve right away. In the first place it seems like more effort, but I am trying to do this lately and the reviews are going through faster then

There is quite some copy-pasting going on here, but this I think you solved in the other PR by unifying those processes right?

File: kratos/python_scripts/assign_vector_by_direction_to_condition_process.py (hunk 1 above)
Comment (loumalouomega) [reply kratos/python_scripts/assign_vector_by_direction_to_condition_process.py]: For the people to be aware

**Comment by loumalouomega** [comment]
> the `assign_scalar_variable_process` and the `assign_scalar_variable_to_conditions_process` can handle the string-defined direction?
> 
That's the idea of thie PR
> With all the cosmetic changes I can barely follow/understand the changes. Can you please make a PR with only the cosmetic changes? This is an important process, we have to be careful
OK
> In general, you can request my review with cosmetic-only PRs, I would approve right away. In the first place it seems like more effort, but I am trying to do this lately and the reviews are going through faster then
> 
> There is quite some copy-pasting going on here, but this I think you solved in the other PR by unifying those processes right?

The unification is a PR that it is in the oven from long time

**Comment by loumalouomega** [comment]
@philbucher Now is easier to check

File: kratos/python_scripts/assign_vector_by_direction_to_condition_process.py (hunk 1 above)
Comment (philbucher) [reply kratos/python_scripts/assign_vector_by_direction_to_condition_process.py]: Aware of what?
=> if I deliberately select this why do I get a warning?

Comment (loumalouomega) [reply kratos/python_scripts/assign_vector_by_direction_to_condition_process.py]: Good point

**Review by philbucher — APPROVED** [review]

**Pull request merged**